### PR TITLE
[benchmark][examples] Enable benchmark test and examples to support BNNS backend on Mac OS.

### PR DESCRIPTION
**Pull request by BruceDai**
@huningxin PTAL, thanks.

### CODE REVIEW

**Comment by huningxin** [comment]
As discussed, we need:
1. set PREFER_SUSTAINED_SPEED by default
2. allow user to specify the preference explicitly
3. if backend throws error, log the error and switch to default backend (WASM).

**Comment by BruceDai** [comment]
@huningxin I updated code following the above solution we discussed, and I also handle the invalid prefer not 'fast' and 'sustained', this also switch to default backend (WASM).
PTAL, thanks.

**Comment by BruceDai** [comment]
Fixes #123 #163

File: benchmark/index.js [hunk 1]
```diff
@@ -298,25 +286,101 @@ document.addEventListener('DOMContentLoaded', () => {
     modelName: 'mobilenet',
     iteration: 0
   }];
-  let webmlAPIConfigurations = [{
+  let webmlAPIConfigurationsAndroid = [{
+    framework: 'Web ML API',
```
Comment (huningxin) [inline benchmark/index.js]: 'WebML API'

File: benchmark/index.js [hunk 2]
```diff
@@ -298,25 +286,101 @@ document.addEventListener('DOMContentLoaded', () => {
...
+    backend: 'NN',
```
Comment (huningxin) [inline benchmark/index.js]: How about let user to choose preference? I may not expose the underlying backend here.

File: benchmark/index.js [hunk 3]
```diff
@@ -298,25 +286,101 @@ document.addEventListener('DOMContentLoaded', () => {
...
+    modelName: 'mobilenet',
+    iteration: 0
+  }];
+  let webmlAPIConfigurationsMacOS = [{
+    framework: 'Web ML API',
```
Comment (huningxin) [inline benchmark/index.js]: ditto

File: examples/mobilenet/MobileNet.js [hunk 4]
```diff
@@ -53,6 +57,22 @@ class MobileNet {
     return 'success';
   }
 
+  _getPrefer() {
+    let prefer = this._nn.PREFER_FAST_SINGLE_ANSWER;
```
Comment (huningxin) [inline examples/mobilenet/MobileNet.js]: Please use PREFER_SUSTAINED_SPEED (MPS) by default

File: examples/posenet/src/PoseNet.js [hunk 5]
```diff
@@ -70,6 +74,21 @@ class PoseNet{
     return 'success';
   }
 
+  _getPrefer() {
```
Comment (huningxin) [inline examples/posenet/src/PoseNet.js]: Can we put the implementation into util.js? So avoid code duplication.

File: examples/squeezenet/SqueezeNet.js [hunk 6]
```diff
@@ -53,6 +57,22 @@ class SqueezeNet {
     return 'success';
   }
 
+  _getPrefer() {
```
Comment (huningxin) [inline examples/squeezenet/SqueezeNet.js]: ditto

File: examples/ssd_mobilenet/SsdMobileNet.js [hunk 7]
```diff
@@ -85,6 +89,22 @@ class SsdMobileNet {
     return 'success';
   }
 
+  _getPrefer() {
```
Comment (huningxin) [inline examples/ssd_mobilenet/SsdMobileNet.js]: ditto

**Comment by huningxin** [comment]
@BruceDai , thanks for the update. I've completed my 1st round of review. Please take a look at my comments.

File: examples/mobilenet/MobileNet.js (hunk 4 above)
Comment (BruceDai) [reply examples/mobilenet/MobileNet.js]: Yes, for MacOS platform the default preference is set as PREFER_SUSTAINED_SPEED (MPS). Here I set  PREFER_FAST_SINGLE_ANSWER default for others platform.

**Comment by BruceDai** [comment]
@huningxin thanks for your review, I will update them soon and ping you review again.

**Comment by BruceDai** [comment]
@huningxin PTAL, thanks.

**Comment by huningxin** [comment]
Looks good. Thanks!

**Pull request merged**